utils/groovy/src/main/java/org/apache/brooklyn/util/groovy/GroovyJavaMethods.java

```diff
@@ -75,6 +76,20 @@
         }
     }
 
+    /**
+     * Alternative implementation of {@link ScriptBytecodeAdapter#isCase(Object, Object)}<br>
+     * Stripped down to work only for caseExpression of type <code>java.lang.Class</code>.<br>
+     * It behaves the same way only for cases when caseExpression <code>java.lang.Class</code> does <b>not</b> implement <code>isCase</code> method.<br>
+     * It goes directly to {@link DefaultGroovyMethods#isCase(Object, Object)} method instead of using Groovy dynamic invocation.<br>
+     * This safes extra operations and locks used in Groovy dynamic invocation. See <a href="https://issues.apache.org/jira/browse/BROOKLYN-424">BROOKLYN-424</a>.
+     */
+    public static boolean scriptBytecodeAdapter_isCase(Object switchValue, Class caseExpression) {
```

Comment:
If we do keep this method, I'd suggest we change the name, since it now has nothing to do with ScriptByteCodeAdapter, maybe call it something like `safeGroovyIsCase`.